src/main/java/com/iabtcf/utils/BitSetIntIterable.java

```diff
@@ -0,0 +1,135 @@
+package com.iabtcf.utils;
+
+/*-
+ * #%L
+ * IAB TCF Core Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.BitSet;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * An implementation of the IntIterable based on BitSet.
+ */
+public class BitSetIntIterable implements IntIterable {
+    private final BitSet bs;
+
+    public static final BitSetIntIterable EMPTY = new BitSetIntIterable(new BitSet());
+
+    public BitSetIntIterable(BitSet bs) {
+        this.bs = bs;
+    }
+
+    @Override
+    public boolean contains(int value) {
+        return bs.get(value);
+    }
+
+    @Override
+    public boolean containsAll(int... source) {
+        for (int i = 0; i < source.length; i++) {
+            if (!contains(source[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public IntIterator intIterator() {
+        return new IntIterator() {
+            int currentIndex = start();
+
+            public int start() {
+                if (bs.isEmpty()) {
+                    return -1;
+                }
+
+                return bs.nextSetBit(0);
+            }
+
+            @Override
+            public boolean hasNext() {
+                return currentIndex != -1;
+            }
+
+            @Override
+            public Integer next() {
+                return nextInt();
```

Comment:
are not we just loosing everything here by boxing to Integer?

Reply:
no, this method is just here to satisfy the Iterable interface. To avoid boxing you'd use com.iabtcf.utils.BitSetIntIterable.intIterator()

Reply:
oh .. wait..